### PR TITLE
Attempt to fix issue #44

### PR DESCRIPTION
**Pull request by yann510**
Looking for some feedback on the code I've wrote, bit of newbie here, first contribution ever :)

-Charts should now resize accordingly to it's parent container
-Legend display isn't always behaving as expected
-On the demo when selecting a chart type, the chart doesn't take it's container space

### CODE REVIEW

**Comment by marjan-georgiev** [comment]
Thanks @yann510! I'll review now.

File: demo/app.ts [hunk 1]
```diff
@@ -1,402 +1,327 @@
...
-        <div *ngIf="chart.options.includes('timeline')">
-          <label>
-            <input type="checkbox" [checked]="timeline" (change)="timeline = $event.target.checked">
-            Timeline
-          </label> <br />
-        </div>
-      </div>
-    </main>
-  `
+  template: `<main>
```
Comment (marjan-georgiev) [inline demo/app.ts]: Could you revert this to it's old formatting please?

File: demo/demo.html [hunk 2]
```diff
@@ -0,0 +1,316 @@
+<main>
```
Comment (marjan-georgiev) [inline demo/demo.html]: I don't see this file being imported anywhere. Let's delete it if we don't use it.

File: package.json [hunk 3]
```diff
@@ -121,5 +116,8 @@
     "webpack-notifier": "^1.3.0",
     "zone.js": "0.6.21",
     "uglify-js": "2.7.3"
+  },
+  "dependencies": {
+    "raw-loader": "^0.5.1"
```
Comment (marjan-georgiev) [inline package.json]: Is this being used anywhere?

File: release/area-chart/area-chart.component.metadata.json [hunk 4]
```diff
@@ -1 +1 @@
..._symbolic":"property","decorators":[{"__symbolic":"call","expression":{"__symbolic":"reference","module":"@angular/core","name":"Input"}}]}],"gradient":[{"__symbolic":"property","decorators":[{"__symbolic":"call","expression":{"__symbolic":"reference","module":"@angular/core","name":"Input"}}]}],"clickHandler":[{"__symbolic":"property","decorators":[{"__symbolic":"call","expression":{"__symbolic":"reference","module":"@angular/core","name":"Output"}}]}],"ngOnChanges":[{"__symbolic":"method"}],"update":[{"__symbolic":"method"}],"getXDomain":[{"__symbolic":"method"}],"getYDomain":[{"__symbolic":"method"}],"getSeriesDomain":[{"__symbolic":"method"}],"getXScale":[{"__symbolic":"method"}],"getYScale":[{"__symbolic":"method"}],"getScaleType":[{"__symbolic":"method"}],"isDate":[{"__symbolic":"method"}],"updateDomain":[{"__symbolic":"method"}],"updateHoveredVertical":[{"__symbolic":"method"}],"hideCircles":[{"__symbolic":"method","decorators":[{"__symbolic":"call","expression":{"__symbolic":"reference","module":"@angular/core","name":"HostListener"},"arguments":["mouseleave"]}]}],"click":[{"__symbolic":"method"}],"trackBy":[{"__symbolic":"method"}],"setColors":[{"__symbolic":"method"}]}}}}
```
Comment (marjan-georgiev) [inline release/area-chart/area-chart.component.metadata.json]: Let's revert all release files please. Release files should be committed only when we release a new version.

File: src/area-chart/area-series.component.ts [hunk 5]
```diff
@@ -31,6 +31,8 @@ export class AreaSeries implements OnChanges {
 
   @Output() clickHandler = new EventEmitter();
 
+
```
Comment (marjan-georgiev) [inline src/area-chart/area-series.component.ts]: Unneeded extra lines.

File: src/bar-chart/series-horizontal.component.ts [hunk 6]
```diff
@@ -7,8 +7,8 @@ import {
   trigger,
   style,
   transition,
-  animate
- } from '@angular/core';
+  animate, NgZone, ElementRef
```
Comment (marjan-georgiev) [inline src/bar-chart/series-horizontal.component.ts]: Unused imports

File: src/common/area.component.ts [hunk 7]
```diff
@@ -45,6 +45,8 @@ export class Area implements OnChanges {
     this.element = element.nativeElement;
   }
 
+
```
Comment (marjan-georgiev) [inline src/common/area.component.ts]: Unneeded extra lines

File: src/common/base-chart.component.ts [hunk 8]
```diff
@@ -1,26 +1,78 @@
...
   }
 
+  setChartSizeBasedOnContainer() {
+    const hostElem = this.chartElement.nativeElement;
+
+    //Get the container dimensions
+    let width = hostElem.parentNode.clientWidth;
+    let height = hostElem.parentNode.clientHeight;
+
+    console.log('container width', width);
```
Comment (marjan-georgiev) [inline src/common/base-chart.component.ts]: Let's remove console logs please.

File: src/common/charts/chart.component.ts [hunk 9]
```diff
@@ -29,23 +29,35 @@ import { Component, Input, OnChanges } from '@angular/core';
...
   title: any;
   legendWidth: any;
   legendType: any;
 
+
+  constructor(private element: ElementRef, private applicationRef: ApplicationRef) {
+
+  }
+
+  ngOnInit(): void {
```
Comment (marjan-georgiev) [inline src/common/charts/chart.component.ts]: Empty implementation.

File: src/common/charts/chart.component.ts [hunk 10]
```diff
@@ -29,23 +29,35 @@ import { Component, Input, OnChanges } from '@angular/core';
...
+    console.log(this);
+  }
+
   ngOnChanges() {
     this.update();
   }
 
+  ngAfterViewInit() {
```
Comment (marjan-georgiev) [inline src/common/charts/chart.component.ts]: Empty implementation.

File: src/common/charts/chart.component.ts [hunk 11]
```diff
@@ -29,23 +29,35 @@ import { Component, Input, OnChanges } from '@angular/core';
...
   @Input() colors;
-
   chartWidth: any;
+
   title: any;
   legendWidth: any;
   legendType: any;
 
+
+  constructor(private element: ElementRef, private applicationRef: ApplicationRef) {
```
Comment (marjan-georgiev) [inline src/common/charts/chart.component.ts]: Are element and applicationRef used anywhere?

**Comment by yann510** [comment]
Thanks for the review, now I make all the required changes and update the pull request?

**Comment by marjan-georgiev** [comment]
Yes, just commit to the same branch and push.

File: release/heat-map/heat-map.module.d.ts [hunk 12]
```diff
@@ -1,2 +1,6 @@
+import { HeatMapCell } from "./heat-map-cell.component";
```
Comment (marjan-georgiev) [inline release/heat-map/heat-map.module.d.ts]: Release files are still here. Please revert them to latest version in the master branch.

**Comment by marjan-georgiev** [comment]
@yann510 could you please revert any change to the files in the `release` folder? Those should only be modified when we release a new version.

Also, please rebase this onto the master branch. I pushed a fix for the legend positioning in there.

**Comment by yann510** [comment]
@marjan-georgiev I've had a little fight with git, I believe everything is fine and the rebase was completed successfully haha

**Comment by marjan-georgiev** [comment]
@yann510 thanks for the updates. 

I am getting several errors testing this on the demo page:

`Uncaught TypeError: Cannot read property 'clientWidth' of null`
`Uncaught TypeError: Cannot read property 'parentNode' of undefined`

**Comment by yann510** [comment]
@marjan-georgiev I have resolved the error mentioned above.

Added the missing resize events on remaining charts (seems like I forgot a few)
NOTE: Pie charts (regular and advanced) if you make it's container width larger, but not it's height, it won't appear bigger as it is a circle, so regular behavior there.

**Pull request merged**